### PR TITLE
Add cfn signals to EC2 products

### PR DESCRIPTION
**Pull request by ahayden**
This change adds a Cloudformation signal counter to common instance templates so that the 'complete' status is reflected in the provisioned product only when all commands in the instance metadata bootstrapping complete successfully. 

This solution works with all current product types.

### CODE REVIEW

File: ec2/sc-ec2-linux-jumpcloud-notebook.yaml [hunk 1]
```diff
@@ -373,6 +383,10 @@ Resources:
           Value: "yes"
         - Key: "PatchGroup"
           Value: "prod-default"
+    CreationPolicy:
+      ResourceSignal:
+        Count: 7 #Must equal the number of calls to cfn-signal during provisioning
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud-notebook.yaml:388]: I actually think we only need 1 signal, the one from `UserData:`.  That signal is sent after all init sections have completed.

File: ec2/sc-ec2-windows-jumpcloud.yaml [hunk 2]
```diff
@@ -238,6 +238,9 @@ Resources:
           Value: "yes"
         - Key: "PatchGroup"
           Value: "prod-default"
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT10M
```
Comment (zaro0508) [inline ec2/sc-ec2-windows-jumpcloud.yaml:243]: i think this may need 20 mins

File: ec2/sc-ec2-linux-jumpcloud-notebook.yaml [hunk 3]
```diff
@@ -373,6 +372,9 @@ Resources:
           Value: "yes"
         - Key: "PatchGroup"
           Value: "prod-default"
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT10M
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud-notebook.yaml:377]: do you think this is enough of a timeout for notebook?

File: ec2/sc-ec2-linux-jumpcloud-notebook.yaml [hunk 4]
```diff
@@ -341,7 +340,7 @@ Resources:
               group: "root"
           commands:
             01_rewrite_apache_conf:
-              command: "/bin/bash /opt/sage/bin/apache_conf_rewrite.sh"
+              command: "/bin/bash /opt/sage/bin/apache_conf_rewrite.sh "
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud-notebook.yaml:343]: is this change intentional?

File: ec2/sc-ec2-linux-jumpcloud.yaml [hunk 5]
```diff
@@ -295,12 +295,11 @@ Resources:
         TagRootVolume:
           commands:
             01_tag_root_disk:
-              command: !Join
-                - ''
-                - - ". /opt/sage/bin/instance_env_vars.sh;"
-                  - "/usr/bin/aws ec2 create-tags --region $AWS_REGION --resources $ROOT_DISK_ID "
-                  - "--tags Key=Name,Value=$EC2_INSTANCE_ID-root Key=cloudformation:stack-name,Value=$STACK_NAME "
-                  - "Key=Department,Value=$DEPARTMENT Key=Project,Value=$PROJECT Key=OwnerEmail,Value=$OWNER_EMAIL"
+              command: !Sub |
```
Comment (zaro0508) [inline ec2/sc-ec2-linux-jumpcloud.yaml:298]: if you refactor this for one file why not refactor for all files?

Comment (ahayden) [reply ec2/sc-ec2-linux-jumpcloud.yaml:298]: There are only a few stings like this., and I've repeated it across the templates that have this string. If we want to refactor to simply all the templates, we can make a new ticket about it.

File: ec2/sc-ec2-linux-jumpcloud-notebook.yaml (hunk 4 above)
Comment (ahayden) [reply ec2/sc-ec2-linux-jumpcloud-notebook.yaml]: Was not!

File: ec2/sc-ec2-windows-jumpcloud.yaml (hunk 2 above)
Comment (ahayden) [reply ec2/sc-ec2-windows-jumpcloud.yaml]: This value was fine during testing, but I can set it to 20.

File: ec2/sc-ec2-linux-jumpcloud-notebook.yaml (hunk 3 above)
Comment (ahayden) [reply ec2/sc-ec2-linux-jumpcloud-notebook.yaml]: The duration was fine during testing, but I've increased it to 20 as I don't think it will harm anything.

File: ec2/sc-ec2-linux-jumpcloud-notebook.yaml (hunk 1 above)
Comment (ahayden) [reply ec2/sc-ec2-linux-jumpcloud-notebook.yaml]: Confirmed
